### ISSUE TITLE
jRuby 9.4 fails to load Celluloid gem

### ISSUE THREAD

**Issue by rykov** [jruby/jruby#7556]
**Environment Information**

* jruby 9.4.0.0 (3.1.0) 2022-11-23 95c0ec159f OpenJDK 64-Bit Server VM 11.0.17+0 on 11.0.17+0 +jit [arm64-darwin]
* Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000 arm64

Other relevant info you may wish to add:
* celluloid gem v0.18.0

**Expected Behavior**

* jRuby 9.3.9.0 successfully loads Celluloid without errors:
```sh
$ jruby -v
jruby 9.3.9.0 (2.6.8) 2022-10-24 537cd1f8bc OpenJDK 64-Bit Server VM 11.0.17+0 on 11.0.17+0 +jit [arm64-darwin]
$ gem install celluloid -v 0.18.0
Successfully installed celluloid-0.18.0
...
$ jruby -e "require 'celluloid'; puts 'ok'"
ok
```

**Actual Behavior**

* jRuby 9.4.0.0 throws `java.lang.ClassCastException` exception:
```sh
$ jruby -v
jruby 9.4.0.0 (3.1.0) 2022-11-23 95c0ec159f OpenJDK 64-Bit Server VM 11.0.17+0 on 11.0.17+0 +jit [arm64-darwin]
$ gem install celluloid -v0.18.0
...
$ jruby -e "require 'celluloid'; puts 'ok'"
LoadError: load error: celluloid/internals/task_set -- java.lang.ClassCastException: class org.jruby.internal.runtime.methods.SynchronizedDynamicMethod cannot be cast to class org.jruby.internal.runtime.methods.IRMethodArgs (org.jruby.internal.runtime.methods.SynchronizedDynamicMethod and org.jruby.internal.runtime.methods.IRMethodArgs are in unnamed module of loader 'bootstrap')
  require at org/jruby/RubyKernel.java:1057
  require at .../jruby-9.4.0.0/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:160
   <main> at .../jruby-9.4.0.0/lib/ruby/gems/shared/gems/celluloid-0.18.0/lib/celluloid.rb:513
  require at org/jruby/RubyKernel.java:1057
  require at .../jruby-9.4.0.0/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:160
   <main> at -e:1
... 7 levels...
```

**Comment by headius** [comment]
That's an interesting one. Investigating.

**Comment by headius** [comment]
Here is one possible fix. Will discuss with @enebo since he did the implementation of this code.

```diff
diff --git a/core/src/main/java/org/jruby/RubyModule.java b/core/src/main/java/org/jruby/RubyModule.java
index 07aa9c1b0b..aa54ac9921 100644
--- a/core/src/main/java/org/jruby/RubyModule.java
+++ b/core/src/main/java/org/jruby/RubyModule.java
@@ -2981,7 +2981,7 @@ public class RubyModule extends RubyObject {
                 if (!method.isNative()) {
                     Signature signature = method.getSignature();
                     if (signature.hasRest() && !signature.hasKwargs()) {
-                        ((IRMethodArgs) method).setRuby2Keywords();
+                        ((IRMethodArgs) method.getRealMethod()).setRuby2Keywords();
                     } else {
                         context.runtime.getWarnings().warn(IRubyWarnings.ID.MISCELLANEOUS, str(context.runtime, "Skipping set of ruby2_keywords flag for ", name, " (method accepts keywords or method does not accept argument splat)"));
                     }
```

**Comment by rykov** [comment]
@headius Thanks for taking a look.

**Comment by enebo** [comment]
This will be fixed by #7561
